tests/functional/api/BackbeatServer.js

```diff
@@ -235,6 +272,333 @@ describe('Backbeat Server', () => {
+                setHash(redisClient, [key], '{}', err => {
+                    assert.ifError(err);
+                    getRequest('/_/crr/failed', (err, res) => {
+                        assert.ifError(err);
+                        assert.deepStrictEqual(res, [{
+                            bucket: 'test-bucket',
+                            key: 'test-key',
+                            versionId: 'test-versionId',
+                            site: 'test-site',
+                        }]);
+                        done();
+                    });
+                });
+            });
+
+            it('should get correct data for GET route: /_/crr/failed when ' +
+            'the hash has been created and there are multiple hash keys',
+            done => {
+                const keys = [
+                    'test-bucket:test-key:test-versionId:test-site',
+                    'test-bucket-1:test-key-1:test-versionId-1:test-site-1',
+                    'test-bucket-2:test-key-2:test-versionId-2:test-site-2',
+                ];
+                setHash(redisClient, keys, '{}', err => {
+                    assert.ifError(err);
+                    getRequest('/_/crr/failed', (err, res) => {
+                        assert.ifError(err);
+                        assert.deepStrictEqual(res, [{
+                            bucket: 'test-bucket',
+                            key: 'test-key',
+                            versionId: 'test-versionId',
+                            site: 'test-site',
+                        }, {
+                            bucket: 'test-bucket-1',
+                            key: 'test-key-1',
+                            versionId: 'test-versionId-1',
+                            site: 'test-site-1',
+                        }, {
+                            bucket: 'test-bucket-2',
+                            key: 'test-key-2',
+                            versionId: 'test-versionId-2',
+                            site: 'test-site-2',
+                        }]);
+                        done();
+                    });
+                });
+            });
+
+            it('should get correct data at scale for GET route: /_/crr/failed',
+            function f(done) {
+                this.timeout(30000);
+                const cmds = [];
+                for (let i = 0; i < 20000; i++) {
+                    const keys = [
+                        `bucket-${i}:key-${i}:versionId-${i}:site-${i}-a`,
+                        `bucket-${i}:key-${i}:versionId-${i}:site-${i}-b`,
+                        `bucket-${i}:key-${i}:versionId-${i}:site-${i}-c`,
+                        `bucket-${i}:key-${i}:versionId-${i}:site-${i}-d`,
+                        `bucket-${i}:key-${i}:versionId-${i}:site-${i}-e`,
+                    ];
+                    /* eslint-disable no-loop-func */
+                    cmds.push(cb => setHash(redisClient, keys, '{}', cb));
```

Comment:
Instead of spawning new commands in a `for` loop then applying them in parallel, you could use `async.timesLimit` and do the `getRequest` directly (`times` gives you the current index as a callback param - see https://caolan.github.io/async/docs.html#timesLimit)